charts/datadog/templates/cluster-agent-deployment.yaml

```diff
@@ -178,14 +178,18 @@ spec:
             value: {{ .Values.clusterAgent.admissionController.mutateUnlabelled | quote }}
           - name: DD_ADMISSION_CONTROLLER_SERVICE_NAME
             value: {{ template "datadog.fullname" . }}-cluster-agent-admission-controller
-          {{- if .Values.clusterAgent.admissionController.configMode }}
           - name: DD_ADMISSION_CONTROLLER_INJECT_CONFIG_MODE
+            {{- if .Values.clusterAgent.admissionController.configMode }}
+            value: {{ .Values.clusterAgent.admissionController.configMode }}
+            {{- else if eq (include "trace-agent-use-uds" .) "true" }}
+            value: socket
+            {{- else if eq (include "trace-agent-use-tcp-port" .) "true" }}
+            value: hostip
+            {{- else }}
             value: {{ .Values.clusterAgent.admissionController.configMode | quote }}
-          {{- if eq .Values.clusterAgent.admissionController.configMode "service" }}
+            {{- end }}
```

Comment:
Can we add a 2 additional test cases: one with both socket and port enabled and another one with apm disabled. I suspect this code path might panic since configMode can be nil.

Reply:
I added the test test cases to `/ci` and they both enable `socket`. I added them as conditions too just to be sure.